templates/msg.h.em

```diff
@@ -166,7 +169,9 @@ void _@(msg_underscored_name)_decode(const CanardRxTransfer* transfer, uint32_t*
 @(ind)(void)bit_ofs;
 @(ind)(void)msg;
 @(ind)(void)tao;
-
+@(ind)if (tao && (transfer->payload_len > @(msg_define_name.upper())_MAX_SIZE)) {
```

Comment:
we need this to be in the function above so we can return true to indicate a bad decode. Note that relying on bit_ofs to detect that we haven't decoded is not sufficient